core/templates/dev/head/collection_editor/CollectionEditorNavbarBreadcrumbDirective.js

```diff
@@ -0,0 +1,66 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Primary controller for the collection editor navbar breadcrumb.
+ *
+ * @author oskar.cieslik@gmail.com (Oskar Cieslik)
+ */
+
+oppia.directive('collectionEditorNavbarBreadcrumbDirective', [function() {
+  return {
+    restrict: 'E',
+    templateUrl: 'inline/collection_editor_navbar_breadcrumb_directive',
+    controller: ['$scope', 'WritableCollectionBackendApiService',
+      'CollectionRightsBackendApiService', 'CollectionObjectFactory',
+      'CollectionUpdateService', 'UndoRedoService', 'warningsData',
+      'routerService', 'CollectionBackendApiService', function(
+      $scope, WritableCollectionBackendApiService,
+      CollectionRightsBackendApiService, CollectionObjectFactory,
+      CollectionUpdateService, UndoRedoService, warningsData, routerService,
+      CollectionBackendApiService) {
+      $scope.collection = null;
```

Comment:
Perhaps it would be ideal to pass in the collection from the parent ``collection_editor`` page, similar to https://github.com/oppia/oppia/blob/2e6b2e60bb6f12fbd3a144238f2d8198ebe5ade6/core/templates/dev/head/collection_editor/CollectionDetailsEditorDirective.js#L25. This way, only ``CollectionEditor`` is responsible for loading the collection and shares it between all directives used.